src/comm.cpp

```diff
@@ -667,13 +667,16 @@ double Comm::get_comm_cutoff()
 
   // always take the larger of max neighbor list and user specified cutoff
 
-  maxcommcutoff = MAX(cutghostuser,neighbor->cutneighmax);
+  if (force->pair)
```

Comment:
I don't think this works.  Say a user does a first run with a long pair cutoff, then a 2nd run with
short pair cutoff.  Then on the 2nd run, this will set maxcommcutoff to the too-big
value from the first run, not the new smaller value?  Which will then be used by compute rdf, etc
b/c it is invoking this Comm method too early.

Reply:
`maxcommcutoff` is a **local** variable. the `get_comm_cutoff()` method computes and returns it. it doesn't "set" anything. it is computed fresh every time. the actual communication cutoff is set when the comm class calls it later. this way, this computation will do the determination of the communication cutoff consistently rather than having to program special cases all over the place (which will then be missing bits and pieces as we tweaks and tune settings).
I have been lobbying for abandoning direct access to class members and replacing it with accessor functions for a long time. this is a perfect case for why this is a good thing.

Reply:
My point is that compute rdf is calling that method too early, before Comm itself
calls it to set the ghost comm distance.  Thus in some cases (like the one I outlined) 
I think it will return a different answer than Comm ends up using.  Leading to a faulty error check in compute rdf against the RDF cutoff.  I think to be safe, compute rdf (or any compute or fix) that needs the ghost cutoff distance can't check it until post-init().

>I have been lobbying for abandoning direct access to class members and replacing it with >accessor functions for a long time. this is a perfect case for why this is a good thing.

For this value, it's a question of when you access it, not how.